### PR TITLE
docs: update replication docs

### PR DESCRIPTION
**Pull request by hagen1778**
Users find that current explanation of replication isn't clear enough for understanding. The modified version of the doc supposed to improve the situation.

Signed-off-by: hagen1778 <roman@victoriametrics.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -490,17 +490,50 @@ It is available in the [helm-charts](https://github.com/VictoriaMetrics/helm-cha
...
+VictoriaMetrics supports application-level replication to provide availability and prevent from data loss when one
+or more vmstorage nodes are unavailable. When replication is enabled, `vminsert` replicates every incoming sample 
+to `-replicationFactor` distinct `vmstorage` nodes. For example, with `-replicationFactor=2` it is safe to drop
```
Comment (f41gh7) [inline README.md:495]: I think, that we should mention, how `replicationFactor` works for `vmselect` component. 

 And also mention case, when replicationFactor == vmstorage count. It's not recommended setup and data loss is possible for this case.

Comment (hagen1778) [reply README.md:495]: > I think, that we should mention, how replicationFactor works for vmselect component.

I think there is no need in mentioning vmselect here.

> And also mention case, when replicationFactor == vmstorage count. It's not recommended setup and data loss is possible for this case.

This is explained on L:515, isn't it?

**Review by valyala — CHANGES_REQUESTED** [review]
The proposed change of the replication docs isn't good enough comparing to the current docs.

Let's start with the list of questions users fail to answer when reading these docs.

**Comment by hagen1778** [comment]
> Let's start with the list of questions users fail to answer when reading these docs.

The PR is based on the following thread https://victoriametrics.slack.com/archives/CGZF1H6L9/p1663167497278479?thread_ts=1663121631.022429&cid=CGZF1H6L9

**Pull request closed without merging**